pkg/testutils/lint/lint_test.go

```diff
@@ -633,6 +633,54 @@ func TestLint(t *testing.T) {
 		}
 	})
 
+	t.Run("TestDisallowedCharacterSequences", func(t *testing.T) {
+		t.Parallel()
+		disallowedCharacterSequences := []string{
+			"\u202A", // LEFT-TO-RIGHT-EMBEDDING
+			"\u202B", // RIGHT-TO-LEFT-EMBEDDING
+			"\u202C", // POP-DIRECTIONAL-FORMATTING
+			"\u202D", // LEFT-TO-RIGHT-OVERRIDE
+			"\u202E", // RIGHT-TO-LEFT-OVERRIDE
+			"\u2066", // LEFT-TO-RIGHT-ISOLATE
```

Comment:
Are the "isolate" characters actually dangerous? I understand the problems with the override characters, but I've also had to use them in a real project to get bidi text to render correctly. The isolate characters were introduced more recently (2013, after the last time I had to deal with these issues) and were supposed to be a safer alternative. 

This is important because banning *all* of these characters tells speakers of RTL languages that their language cannot be represented properly in source code. That may be acceptable in CockroachDB since it is already our policy that source code and comments be in English, but it would not be good if this rule spread into compilers and other language-wide tools. If we could narrow the block list to only the override and embedding characters while allowing isolates, that would be a better outcome since it could address the security concerns while still allowing all languages to be properly represented (I think. I can't read any RTL languages so there's still quite a bit of conjecture here).

Reply:
The trojan source paper (https://trojansource.codes/trojan-source.pdf) does have an example of malicious use of right-to-left isolate. (maybe first-strong-isolate is still safe, but I'm not sure).

I also suspect that there's a way to combine homoglyph attacks with bidi manipulation without any explicit RTL control characters. For example, the arabic date separator U+060D can pass for a comma in some fonts and it is flagged for RTL mode.